django_crate/backend/schema.py

```diff
@@ -80,109 +98,65 @@ def create_model(self, model):
-                    self.deferred_sql.extend(autoinc_sql)
-            """
-
-        # CRATE OPTIONS
+
+        # Make the table
+        sql = self.sql_create_table % {
+            "table": self.quote_name(model._meta.db_table),
+            "definition": ", ".join(column_sqls),
+            "clustered_by": self._clustered_by_sql(model, params),
+            "partitioned_by": self._partitioned_by_sql(model, params),
+            "settings": self._table_settings_sql(model, params),
+        }
+
+        # Prevent using [] as params, in the case a literal '%' is used in the definition
+        self.execute(sql, params or None)
+        # wait for shards to start
+        # let's do this optimistically ...
+        time.sleep(2)
+
+    def _clustered_by_sql(self, model, params):
         clustered_by = getattr(model._meta, "clustered_by", None)
         number_of_shards = getattr(model._meta, "number_of_shards", None)
-        clustering = ""
+        sql = ''
         if clustered_by or number_of_shards:
-            clustering = " CLUSTERED"
+            clustering = ' CLUSTERED'
             if clustered_by:
-                clustering += " BY ({0})".format(self.quote_name(clustered_by))
+                sql += " BY ({})".format(self.quote_name(clustered_by))
             if number_of_shards:
-                clustering += " INTO {0} SHARDS".format(number_of_shards)
+                sql += " INTO {} SHARDS".format(number_of_shards)
+        return sql
 
-        partitioned = ""
-        partitioned_by = getattr(model._meta, "partitioned_by", [])
+    def _partitioned_by_sql(self, model, params):
+        partitioned_by = getattr(model._meta, "partitioned_by", None)
         if partitioned_by:
-            partitioned = " PARTITIONED BY ({0})".format(
-                ", ".join((self.quote_name(partition) for partition in partitioned_by))
+            return " PARTITIONED BY ({})".format(
+                ", ".join((self.quote_name(p) for p in partitioned_by))
             )
+        return ''
 
-        number_of_replicas = getattr(model._meta, "number_of_replicas", None)  # TODO: move to global default
+    def _table_settings_sql(self, model, params):
+        number_of_replicas = getattr(model._meta, "number_of_replicas", None)
         refresh_interval = getattr(model._meta, "refresh_interval", None)
-        table_settings = ""
-        if number_of_replicas or refresh_interval:
-            table_settings = "WITH ("
-            table_settings += "number_of_replicas=?"
+        table_settings = []
+
+        if number_of_replicas is not None:
+            table_settings.append("number_of_replicas=?")
             params.append(number_of_replicas)
-            table_settings += ", refresh_interval=?"
+        if refresh_interval is not None:
+            table_settings.append("refresh_interval=?'")
             params.append(refresh_interval)
-            table_settings += ")"
-
 
-        sql = self.sql_create_table % {
```

Comment:
FYI I haven't tested any of this, it is leftover bits from crate-django project that I absorbed, I didn't exactly want to remove it, but I have no idea if any of this worked at all, and/or if it works now :)